plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/kotlin/KSPropertyDeclarationImpl.kt

```diff
@@ -35,8 +36,17 @@ class KSPropertyDeclarationImpl private constructor(val ktProperty: KtProperty)
         }
     }
 
+    private fun shouldCreateSyntheticAccessor(): Boolean {
+        return !this.isPrivate()
+                && (
+                !this.isLocal() && !this.modifiers.contains(Modifier.ABSTRACT)
+                        && (this.parentDeclaration as? KSClassDeclaration)?.classKind != ClassKind.INTERFACE
+                        || ((this.parentDeclaration as? KSClassDeclaration)?.classKind == ClassKind.INTERFACE && ktProperty.accessors.isNotEmpty())
+                )
```

Comment:
nit: In other words,
`not_private && (not_local && not_abstract && not_interface || (is_interface && has_accessor))`
Is `is_interface` needed?